java/libs/src/main/java/org/physical_web/collection/PhysicalWebCollection.java

```diff
@@ -211,4 +216,65 @@ public void jsonDeserialize(JSONObject jsonObject) throws PhysicalWebCollectionE
       addMetadata(pwsResult);
     }
   }
+
+  /**
+   * Return a list of PwPairs sorted by rank in descending order.
+   * @param dedupSiteUrls whether or not to dedup PwPairs by result urls.
+   * @param minRank the minimum rank to keep in the list.
+   * @return a sorted list of PwPairs.
+   */
+  public List<PwPair> getPwPairsSortedByRank(boolean dedupSiteUrls, double minRank) {
```

Comment:
Why do we return `PwPair` here?  I think we should just return `PwsResult` list.

Returning the UrlDevice may have uses for debugging, but we conceivably may want to return PwsResult's which come from sources other than UrlDevices.